libcxx/utils/libcxx/test/params.py

```diff
@@ -57,6 +57,8 @@
     # Disable warnings for extensions used in C++03
     "-Wno-local-type-template-args",
     "-Wno-c++11-extensions",
+    "-Wno-c++14-extensions",
+    "-Wno-c++17-extensions",
```

Comment:
Also worthy of attention.

Reply:
Also we would need to ensure that if users are using `-isystem` to include libc++, they never get these extension warnings. If they started to for some reason, for example an extension is used in a default argument and that triggers a diagnostic from the user's code despite `-isystem`, then we should catch that. IDK if that's even possible, but I think it would be safer to find a way not to disable extension warnings for the whole test suite.

It doesn't have to be done as part of this review, but I think it's important enough that we'd want to do it as a follow-up: shipping a bug of that sort could be quite disruptive and it would be even more embarrassing since it seems really easy to catch from a user's perspective.

Reply:
`-isystem` suppresses warnings for the headers it pulls in, so warnings will be suppressed unless we're exposing things via macros (which we aren't currently doing). I certainly wouldn't consider any non-macro exposure of a warning from `-isystem` to be a libc++ bug.

Having said that, this is the sort of test that Clang may want to pass on to libraries to detect. @AaronBallman, any insight here?

Reply:
Some diagnostics will appear in system headers, we mark them specially with `ShowInSystemHeader` in our diagnostics files, like: https://github.com/llvm/llvm-project/blob/1ebe73821f4cefa48d7e3d24e62303412ab9ad25/clang/include/clang/Basic/DiagnosticSemaKinds.td#L7879

I don't think that will matter for your needs, but it's still something to keep in mind regarding suppression mechanisms. If a diagnostic that isn't marked `ShowInSystemHeader` is appearing in a system header, that's almost certainly a Clang bug.

FWIW, we also suppress diagnostics coming from system macros: https://github.com/llvm/llvm-project/blob/1ebe73821f4cefa48d7e3d24e62303412ab9ad25/clang/lib/Basic/DiagnosticIDs.cpp#L577